### ISSUE TITLE
Component: OverlayPanel should include a11y patterns 

### ISSUE THREAD

**Issue by NateRadebaugh** [primefaces/primeng#13005]
### Describe the feature you would like to see added

`<p-overlayPanel` should have proper accessibility

I know there's an existing documentation that outlines the desired functionality and it's mentioned that it's a goal for Q2, but I'm wondering if there's a checklist of if PRs would be helpful.

https://primeng.org/overlaypanel

### Describe the solution you'd like

Implement the outlined accessibility:
https://primeng.org/overlaypanel

Something like the following can be achieve in user-land but it would great if this was all abstracted away and handled by `p-overlayPanel` a bit more natively.

1. Add `pFocusTrap` to the contents of the first div inside of `pTemplate`
2. Get a ref to the `#overlayPanel` and the `#trigger`
3. When showing/hiding the `p-overlayPanel`, we `focusOverlay()` or `focusTrigger()`

```html
<p-overlayPanel
  #overlayPanel
  (onShow)="focusOverlay()"
  (onHide)="focusTrigger()"
>
  <ng-template pTemplate>
    <div
      *ngIf="overlayPanel.overlayVisible"
      pFocusTrap
    >
   ...
    </div>
  </ng-template>
</p-overlayPanel>

<div #trigger>
   <button type="button" (click)="toggle($event)">Toggle</button>
</div>
```

4. Add a `focusFirst` helper, possibly in a `focus.service.ts`
```ts
function focusFirst(element: HTMLElement) {
    const focusableElements = element.querySelectorAll(
      'button, [href], input, select, textarea, [tabindex]:not([tabindex="-1"])'
    ) as NodeListOf<
      | HTMLButtonElement
      | HTMLAnchorElement
      | HTMLInputElement
      | HTMLSelectElement
      | HTMLTextAreaElement
    >;
    if (focusableElements.length) {
      focusableElements[0]!.focus();
      return true;
    }

    return false;
}
```

5. Use `@ViewChild()` to get the HTML element refs:
```ts
@ViewChild('overlayPanel') overlayPanel: OverlayPanel | undefined;
@ViewChild('trigger') trigger: ElementRef<HTMLDivElement> | undefined;
```

7. Add event handlers for toggle/hide/show/focus*
```ts
  toggle(event: Event) {
    event.preventDefault();

    if (this.disabled) {
      return;
    }

    if (this.overlayPanel) {
      if (this.overlayPanel.overlayVisible) {
        this.hide();
      } else {
        this.overlayPanel.show(event);
      }
    }
  }

  hide() {
    this.overlayPanel?.hide();
  }

  focusTrigger() {
    const el = this.trigger?.nativeElement as HTMLElement | undefined;
    if (el) {
      this.focusService.focusFirst(el);
    }
  }

  focusOverlay() {
    const el = this.overlayPanel?.container;
    if (el) {
      this.focusService.focusFirst(el);
    }
  }
```

8. Respond to `escape` key being pressed when the overlay is open
```ts
  @HostListener('document:keydown.escape', ['$event'])
  onEscapeKeyPressed(event: KeyboardEvent): void {
    if (this.overlayPanel?.overlayVisible) {
      this.hide();
    }
  }
```

### Describe alternatives you have considered

I could implement this myself in user-land but that's less good.

**Comment by mertsincan** [comment]
Hi,

So sorry for the delayed response! Improvements have been made to many components recently, both in terms of performance and enhancement. Therefore, this improvement may have been developed in another issue ticket without realizing it. You can check this in the documentation. If there is no improvement on this, can you open a new issue so we can include it in our roadmap?

Thanks a lot for your understanding!
Best Regards,
